src/guiFormSpecMenu.cpp

```diff
@@ -3297,7 +3297,8 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 	if (event.EventType==EET_KEY_INPUT_EVENT) {
 		KeyPress kp(event.KeyInput);
 		if (event.KeyInput.PressedDown && ( (kp == EscapeKey) ||
-				(kp == getKeySetting("keymap_inventory")) || (kp == CancelKey))) {
+				((m_client != NULL) && (kp == getKeySetting("keymap_inventory"))) ||
+				(kp == CancelKey))) {
```

Comment:
Could you please bundle the similar checks and add them after eachother? i.e. `(kp == EscapeKey) || (kp == CancelKey) ||`
IMO, easier to read the logic this way.